### PR TITLE
Add StackGAN-v1 model implemented in Tensorflow

### PR DESCRIPTION
**Pull request by Divyanshu23**
The results of the model and its layered architecture aren't pushed currently because it is taking a lot of time to train and my training quota on colab is exceeded.

### CODE REVIEW

**Comment by SomTambe** [comment]
https://gist.github.com/mrrajatgarg/7e9b2bf74ebebaaa9ba22e0e654cc7aa
This source seems to be quite the same as one of your code files.
Care to explain?

**Comment by Divyanshu23** [comment]
> https://gist.github.com/mrrajatgarg/7e9b2bf74ebebaaa9ba22e0e654cc7aa
> This source seems to be quite the same as one of your code files.
> Care to explain?

Yes it certainly is "for the time being." About 70-80% of the implementation has been taken from that source because of trying to implement and understand the whole model in the last two days of the deadline.

**Comment by namangup** [comment]
Add results of your training. Loss / Metrics etc.

**Comment by Divyanshu23** [comment]
Updated all the things as required:-
Key Changes:-
1) Rewritten the model at several places.
2) Modified the structure of Generator and Discriminator.
3) Remove unnecessary architecture is stage1_mode.py and stage2_model.py
4) Changed the loss function of Stage2 generator.
5) Updated the metrics in README.md
